dashboard/src/components/Filter/CodeBlock.tsx

```diff
@@ -21,10 +32,130 @@ interface IHighlightedCode {
   errorCount: number;
 }
 
+interface CodeComponentProps {
+  fullscreen?: boolean;
+}
+
+const CodeBlockDialog = ({
+  CodeComponent,
+}: {
+  CodeComponent: ({ fullscreen }: CodeComponentProps) => JSX.Element;
+}): JSX.Element => {
+  return (
+    <div className="relative">
+      <Dialog>
+        <DialogTrigger asChild className="p-2">
+          <Button
+            variant="outline"
+            className="absolute right-[10px] top-[10px] opacity-50 hover:opacity-100"
+          >
+            <GoScreenFull size={25} />
+          </Button>
+        </DialogTrigger>
+
+        <CodeComponent />
+
+        <DialogContent className="sm:max-w-[70%]">
+          <DialogHeader>
+            <DialogTitle>
+              <FormattedMessage id="global.logs" />
+            </DialogTitle>
+          </DialogHeader>
+          <CodeComponent fullscreen />
+        </DialogContent>
+      </Dialog>
+    </div>
+  );
+};
+
+const Code = ({
+  highlightedCode,
+  className,
+  footer,
+}: {
+  highlightedCode: string;
+  className?: string;
+  footer: JSX.Element;
```

Comment:
This footer could be optional, avoiding having to pass an empty fragment as it is done in CodeBlock.tsx line 232,
```
footer={
  highlightedCode.highlightCount > 0 ? (
    HighlightCounts({
      highlightedCode,
      highlightsClassnames: highlightsClassnames,
    })
  ) : (
    <></>
  )
}
```